### PR TITLE
`identity`: add values from import_format in `identitySchema` not in properties list

### PR DESCRIPTION
**Pull request by BBBmau**
We run into a case in several resources such as dataset where a value within `import_format` is at times a value that is not included in properties list resulting in a value such as `dataset_id` not being included in identity schema. PR resolves this by creating a new Type for the missing value in properties:
```golang
	for _, field := range importFormat {
		if includedValues[field] || field == "project" || field == "zone" || field == "region" {
			continue
		}
		props = append(props, &Type{Name: field, Type: "string", Required: true})
		includedValues[field] = true
	}
```
also included a unit test that replicates this specific edge case

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:none

```

### CODE REVIEW

**Comment by shuyama1** [comment]
Sorry for missing this from my review lists after vacation.

/gcbrun

File: mmv1/products/bigquery/Dataset.yaml [hunk 1]
```diff
@@ -43,8 +43,6 @@ custom_code:
   pre_read: 'templates/terraform/pre_read/bigquery_dataset.go.tmpl'
 custom_diff:
   - 'customCollationDiff'
-# exluding resource_identity due to dataset_id field
```
Comment (shuyama1) [inline mmv1/products/bigquery/Dataset.yaml:46]: I'm surprised by how the generator works here. I can't figure out why `dataset_id` is missing from the properties. The resource is a few years old, so it's a bit hard to trace the history.

**Review by shuyama1 — COMMENTED** [review]
Sorry for the delay. This fell off my radar somehow

Could you confirm if all the generated fields in [downstreams](https://github.com/GoogleCloudPlatform/magic-modules/pull/17759#issuecomment-4568930453) are expected? I'm seeing fields like `name` for `bigquery_dataset_access` that didn't exist in the original resource or import format.

File: mmv1/products/bigquery/Dataset.yaml (hunk 1 above)
Comment (BBBmau) [reply mmv1/products/bigquery/Dataset.yaml:46]: after some more investigation i found that the dataset_id was not being included due to it being a nested field set wtihim `datasetReference` where the `nestedObject` itself is marked as `flatten_object: true` which moves the nested values to root upon rendering the provider.

I had taken time to reply since it seemed odd to have affected other existing resources. I've made the change of including `flatten_object` values as part of the identityProperties check in this PR:
- https://github.com/GoogleCloudPlatform/magic-modules/pull/17980

will be closing this and assigning you the new PR as it's solving the same as what was presented here.
